### PR TITLE
Add option --use-utc-timezone

### PR DESCRIPTION
**Pull request by jumapico**
Make timestamps reproducible adding the flag --use-utc-timezone.
While working on the "reproducible builds" effort [1](https://wiki.debian.org/ReproducibleBuilds), we have noticed
that some packages that use asciidoc for build documentation uses the
current user timezone, making that the package can't be built
reproducibly.
With the new flag --use-utc-timezone that uses the UTC
timezone instead of the local one we can make the generation of the
documentation reproducible.

### CODE REVIEW

**Comment by jumapico** [comment]
Please, consider the commit 0e79d28.
It rewrites the change, using now a global variable for switch between localtime or gmtime.

File: doc/asciidoc.1.txt [hunk 1]
```diff
@@ -95,6 +95,10 @@ OPTIONS
     The *--theme* option is also used to manage theme plugins (see
     <<X1,*PLUGIN COMMANDS*>>).
 
+*--use-utc-timezone*::
+    Use the UTC timezone for show dates and times. Afects docdate,
```
Comment (elextr) [inline doc/asciidoc.1.txt]: Affects

...timezone to show ...

**Comment by elextr** [comment]
Couple of english/spelling corrections, then could you squash the commits, history doesn't need to know the failed attempt :)

**Comment by jumapico** [comment]
Thanks for your review!
I hope that now the pull request is fine.

**Comment by mojavelinux** [comment]
Perhaps the option should simply be `use-utc`. Isn't using UTC understood that it's specifying the timezone?

**Comment by elextr** [comment]
Looking at your initial use-case, this isn't going to help.  Ok, times will be in the same time zone, but each call to gmtime() will get a different value.  If you want to generate reproducable builds you need to either exclude times from the documents or do something like the [mock local time](https://github.com/asciidoc/asciidoc/blob/master/tests/testasciidoc.py#L79) used in testing.

**Comment by aerostitch** [comment]
@jumapico Would #7 solve your initial use case?
It brings the ability to either remove the "last updated" field or give set it to the value of the date field in set in the header. (see also https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=656736)

**Comment by andersk** [comment]
There is now a [SOURCE_DATE_EPOCH environment variable](https://wiki.debian.org/ReproducibleBuilds/TimestampsProposal). I propose that if SOURCE_DATE_EPOCH is set, AsciiDoc should automatically (1) use UTC for all timestamps and (2) replace all timestamps with min(timestamp, SOURCE_DATE_EPOCH). Then projects using AsciiDoc can be reproducible without being patched to pass extra flags.

**Comment by elextr** [comment]
#9 should allow reproducable builds by allowing you to select not showing a date, or using the revdate from the header instead of the file date/time.

**Comment by andersk** [comment]
@elextr #9 _allows_ them, but respecting SOURCE_DATE_EPOCH would make them work by default. When you’re trying to make a Linux distribution with 43000 packages reproducible, the difference is important.

**Comment by elextr** [comment]
@andersk ok, well, if a pull request is provided it can be looked at.

**Comment by andersk** [comment]
I submitted a pull request for SOURCE_DATE_EPOCH as #106.

**Comment by aerostitch** [comment]
@jumapico It seems this repo will no longer accept PR as the latest release has been cut a while ago but https://github.com/asciidoc/asciidoc-py3 would.
If you still want this PR integrated in asciidoc, do you mind pushing it there please?
Once done I think you can close this one.
Thanks!

**Comment by jumapico** [comment]
@aerostitch @mojavelinux : 

I think that this PR is obsoleted by the usage of SOURCE_DATE_EPOCH and the change in #9.
The initial intent was create a *reproducible build*, and with the previous changes it can be done.

Thanks all.

**Pull request closed without merging**